apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java

```diff
@@ -679,7 +679,7 @@ public void downloadBlob(String container, String name, File destination, Execut
 
       ListeningExecutorService listeningExecutor = MoreExecutors.listeningDecorator(executor);
       RandomAccessFile raf = null;
-      File tempFile = new File(destination.getName() + "." + UUID.randomUUID());
+      File tempFile = new File(destination + "." + UUID.randomUUID());
```

Comment:
Given "/tmp/foo" `getName` returns "foo", ensuring the file will be relative to wherever the user launched the jclouds application.  With this change jclouds will create the temporary file in the same directory as the user specified.